scripts/update_appup.escript

```diff
@@ -258,14 +258,39 @@ find_base_appup_actions(App, PrevVersion) ->
         end,
     {ensure_version(PrevVersion, Upgrade), ensure_version(PrevVersion, Downgrade)}.
 
-merge_update_actions(App, Changes, Vsns) ->
+merge_update_actions(App, Changes, Vsns, PrevVersion) ->
     lists:map(fun(Ret = {<<".*">>, _}) ->
                       Ret;
                  ({Vsn, Actions}) ->
-                      {Vsn, do_merge_update_actions(App, Changes, Actions)}
+                      case is_skipped_version(App, Vsn, PrevVersion) of
+                          true ->
+                              log("WARN: ~p has version ~s skipped over?~n", [App, Vsn]),
+                              {Vsn, Actions};
+                          false ->
+                              {Vsn, do_merge_update_actions(App, Changes, Actions)}
+                      end
               end,
               Vsns).
 
+%% say current version is 1.1.2, and the compare baes is version 1.1.1, but there is a 1.1.2 in appup
+%% we may skip merging instructions for 1.1.2 (it's never used)
+is_skipped_version(App, Vsn, PrevVersion) when is_list(Vsn) andalso is_list(PrevVersion) ->
+    case is_app_external(App) andalso parse_version_number(Vsn) of
+        {ok, VsnTuple} ->
+            case parse_version_number(PrevVersion) of
+                {ok, PrevVsnTuple} ->
+                    VsnTuple > PrevVsnTuple;
```

Comment:
Q: won't this skip all external lib upgrades?

Reply:
It only skips the current App (being scanned), the current version (in the `lists:map` call at line 262)

Reply:
If the current version of the external app is skipped, it won't report if there are missing actions.  Isn't that bad for spotting problems?
From what I understood, `VSN` is the version from the appup item.  So, this will skip all versions above that used in the base release, essentially skipping any checks for external apps (that can have their version parsed correctly). :thinking:

Reply:
the 'current' is actually one of the old versions being enumerated by `lists:map`

Reply:
* `Vsn` in this function arg is one of the old versions being enumerated by `lists:map` in `merge_update_actions`
  (The list is from the current version's appup file.)
* `PrevVersion` is the version being used in the upgrade base.